modules/sov-ibc-transfer/src/context.rs

```diff
@@ -203,25 +194,26 @@ where
         coin: &PrefixedCoin,
         memo: &Memo,
     ) -> Result<(), TokenTransferError> {
-        let token_id_buf = general_purpose::STANDARD_NO_PAD
-            .decode(memo.as_ref())
-            .map_err(|e| TokenTransferError::Other(format!("Failed to decode memo: {e}")))?;
+        if !memo.as_ref().is_empty() {
+            return Err(TokenTransferError::Other(
+                "Memo must be empty when escrowing tokens".to_string(),
+            ));
+        }
 
-        let token_id = BorshDeserialize::deserialize(&mut token_id_buf.as_slice())
-            .map_err(|e| TokenTransferError::Other(format!("Failed to decode memo: {e}")))?;
+        let token_id = TokenId::from_str(&coin.denom.to_string()).map_err(|e| {
+            TokenTransferError::InvalidCoin {
+                coin: coin.to_string(),
+            }
+        })?;
 
-        let token_name = self
+        if let Some(token_name) = self
             .ibc_transfer
-            .bank
-            .get_token_name(&token_id, &mut self.working_set.borrow_mut())
-            .ok_or(TokenTransferError::Other(format!(
-                "No token with address {token_id}",
-            )))?;
-
-        if token_name != coin.denom.to_string() {
-            return Err(TokenTransferError::InvalidCoin {
-                coin: coin.to_string(),
-            });
+            .minted_token_id_to_name
+            .get(&token_id, *self.working_set.borrow_mut())
+        {
+            return Err(TokenTransferError::Other(
+                format!("Token with ID '{token_id}' is an IBC-created token and cannot be escrowed. Use '{token_name}' as denom for sending back an IBC token to the source chain"),
```

Comment:
This seems like a long line. You can break the line into multiline.

Example of usage:
```rs
fn main() {
    let st = "hello \
    world";
    println!("{st:?}"); // prints "hello world"
}
```

Reply:
[b927153](https://github.com/informalsystems/sovereign-ibc/pull/133/commits/b9271531b1b259510603b1c0048725a82dceea78)